core/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -622,4 +624,53 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeBoolean(value);
         }
     }
+
+    public void testWriteMapWithConsistentOrder() throws IOException {
+        Map<String, Object> map = new HashMap<>();
+        map.put("gWrgS", randomAsciiOfLength(5));
```

Comment:
I'm not super comfortable relying on the internals of HashMap for this. I wonder if we could get away instead with two `TreeMap`s, one with natural ordering and one with `Comparator.reverse()`.